types.go

```diff
@@ -46,6 +45,17 @@ type Param struct {
 	IsBound bool `json:"isBound,omitempty" yaml:"isBound,omitempty"` // I.e. is a struct reference
 }
 
+type BodyParam struct {
+	Name        string `json:"name,omitempty" yaml:"name,omitempty"`
+	Field       Field  `json:"type,omitempty" yaml:"type,omitempty"`
+	ContentType string `json:"contentType,omitempty" yaml:"contentType,omitempty"`
+	IsRequired  bool   `json:"isRequired,omitempty" yaml:"isRequired,omitempty"`
+	IsArray     bool   `json:"isArray,omitempty" yaml:"isArray,omitempty"`
+	IsMap       bool   `json:"isMap,omitempty" yaml:"isMap,omitempty"`
+
+	IsBound bool `json:"isBound,omitempty" yaml:"isBound,omitempty"` // I.e. is a struct reference
+}
+
```

Comment:
Did the struct embedding idea not work?

Reply:
Unfortunately not, I would have mean initialising the embedded struct separately every time, I can do that but actually I think we should review what is in each param struct anyway.